src/app/frontend/deploy/deployfromsettings_controller.js

```diff
@@ -183,6 +188,30 @@ export default class DeployFromSettingsController {
   }
 
   /**
+   * @returns {!angular.$q.Promise}
+   * @export
+   */
+  updateAvailableProtocols() {
+    let defer = this.q_.defer();
+    /** @type {!backendApi.ProtocolsSpec} */
+    let protocolsQuery = {isExternal: this.isExternal};
+
+    this.protocolsResource.get(
+        protocolsQuery,
+        (protocols) => {
+          defer.resolve(protocols);
+          this.log_.info('Protocols successfuly retrieved: ', protocols.protocols);
+          this.protocols = protocols.protocols;
+          this.portMappings = [this.newEmptyPortMapping_(this.protocols[0])];
```

Comment:
This line causes protocol to be cleared out when I have already something filled in. Can you fix this?